corehq/apps/reports/standard/users/reports.py

```diff
@@ -150,6 +151,18 @@ def rows(self):
         for record in records:
             yield _user_history_row(record, self.domain, self.timezone)
 
+    @classmethod
+    @memoized
+    def get_location_name(self, location_id):
```

Comment:
this should be `cls` instead of `self` if it's a classmethod, but for the memoization to clean itself up properly, it needs to be an instance method.  This will require reorganizing the code below to make it possible.

`@memoized` on a classmethod will follow the lifecycle of the class, meaning it'll persist for the duration of the django process, causing a memory leak.  If you put it on an instance method, it'll persist only as long as the instance, which is cleaned up at the end of the request.

Reply:
I see, I'll look at reworking to make instance method work.